packages/@expressive-code/plugin-collapsible-sections/src/ast.ts

```diff
@@ -0,0 +1,54 @@
+import { type ElementContent } from 'hast'
+import { Section } from './utils'
+import { h, s } from 'hastscript'
+import { collapsibleSectionClass } from './styles'
+
+/**
+ * Text to display in the <summary> of a collapsed section
+ * Can contain the placeholder '{count}'
+ */
+export type SectionText = string
+
+/**
+ * Transforms a list of line ASTs into one containing the provided collapsible sections
+ */
+export function sectionizeAst({
+	lines,
+	sections,
+	text = '{count} collapsed lines',
+}: {
+	lines: ElementContent[]
+	sections: Section[]
+	text?: SectionText | undefined
+}): ElementContent[] {
```

Comment:
```suggestion
export function sectionizeAst({ lines, sections, text }: { lines: ElementContent[]; sections: Section[]; text: (count: number) => string }): ElementContent[] {
```

Text would be an interpolation function in this case.